src/main.rs

```diff
@@ -37,29 +37,32 @@ struct Args {
     #[arg(short, long, action = clap::ArgAction::Count)]
     pub verbose: u8,
 
-    /// Start as a daemon listening on a specific socket.
+    /// Start as a daemon listening on a specific socket
     ///
     /// If the socket address is not given, defaults to 127.0.0.1:3232
     #[arg(short, long)]
     pub listen: Option<Option<PartialAddr<InetHost, DEFAULT_PORT>>>,
 
-    /// Path to an identity (key) file.
-    #[arg(short, long)]
+    /// Path to an identity (key) file
+    #[arg(short, long, require_equals = true)]
     pub id: Option<PathBuf>,
 
-    /// SOCKS5 proxy, as IPv4 or IPv6 socket. If port is not given, it defaults
-    /// to 9050.
-    #[arg(short = 'p', long, conflicts_with = "listen")]
-    pub proxy: Option<PartialAddr<InetHost, DEFAULT_SOCKS5_PORT>>,
+    /// SOCKS5 proxy, as IPv4 or IPv6 socket
+    ///
+    /// If port is not given, defaults to 9050.
+    #[arg(short = 'p', long, conflicts_with = "listen", require_equals = true)]
+    pub proxy: Option<Option<PartialAddr<InetHost, DEFAULT_SOCKS5_PORT>>>,
```

Comment:
This is how proxy argument is passed, such that it works as
`nsh --proxy` - meaning `use 127.0.0.1:9050` as proxy
`nsh --proxy=xx.xx.xx.xx` meaning `use custom proxy`
`nsh` with no proxy -  connect directly